File: app/src/main/java/com/example/notes/CardsSource.java
package com.example.notes;

import java.util.ArrayList;

public interface CardsSource {
    CardData getCardData(int position);

    int size();

    void deleteCardData(int position);

    void updateCardData(int position, CardData cardData);

    void addCardData(CardData cardData);

    void clearCardData();

    void setNewData(ArrayList<CardData> dataSource);

    ArrayList<CardData> getCardData();
}

File: app/src/main/java/com/example/notes/CardsSource.kt
package com.example.notes

import com.example.notes.CardData
import java.util.ArrayList

interface CardsSource {
    fun getCardData(position: Int): CardData?
    fun size(): Int
    fun deleteCardData(position: Int)
    fun updateCardData(position: Int, cardData: CardData?)
    fun addCardData(cardData: CardData?)
    fun clearCardData()
    fun setNewData(dataSource: ArrayList<CardData?>?)
    val cardData: ArrayList<CardData?>?
}